packages/blade/src/components/Chip/Chip.stories.tsx

```diff
@@ -0,0 +1,564 @@
+                Automated Payment Links
+              </ChipComponent>
+              <ChipComponent value="wallet" icon={SmartphoneIcon}>
+                Wallet on My App
+              </ChipComponent>
+              <ChipComponent value="offers" icon={OffersIcon}>
+                Offer discounts, Pay Later & EMI options
+              </ChipComponent>
+            </ChipGroupComponent>
+          </Box>
+        </Box>
+      ))}
+    </Box>
+  );
+};
+
+export const AllChipSizes = AllChipSizesTemplate.bind({});
+AllChipSizes.storyName = 'All Sizes';
+
+const ChipIntentsTemplate: ComponentStory<typeof ChipGroupComponent> = (args) => {
+  return (
+    <Box display="flex" flexDirection="column">
+      <Text size="large" weight="bold" marginBottom="spacing.3">
+        Is the result helpful?
+      </Text>
+
+      <ChipGroupComponent defaultValue="yes" {...args}>
+        <ChipComponent intent="positive" value="yes" icon={ThumbsUpIcon}>
+          Yes
+        </ChipComponent>
+        <ChipComponent intent="negative" value="no" icon={ThumbsDownIcon}>
+          No
+        </ChipComponent>
+      </ChipGroupComponent>
+    </Box>
+  );
+};
+
+export const ChipWithIntent = ChipIntentsTemplate.bind({});
+ChipWithIntent.storyName = 'With Intent';
+ChipWithIntent.args = {
+  accessibilityLabel: 'Is the result helpful?',
+};
+
+const TextTransformationTemplate: ComponentStory<typeof ChipGroupComponent> = ({
+  children,
+  ...args
+}) => {
+  const chipValues = ['Proprietorship', 'Public', 'Small Business'];
+  return (
+    <Box>
+      <Text marginBottom="spacing.3" marginTop="spacing.3">
+        Select Business type:
+      </Text>
+
+      <ChipGroupComponent defaultValue="Proprietorship" {...args}>
+        {chipValues.map((chipValue: string) => (
+          <ChipComponent key={chipValue} value={chipValue}>
+            {chipValue.toUpperCase()}
+          </ChipComponent>
+        ))}
+      </ChipGroupComponent>
+    </Box>
+  );
+};
+
+export const TextTransformationUppercase = TextTransformationTemplate.bind({});
+TextTransformationUppercase.storyName = 'Text Transformation (Uppercase)';
+
+export const chipRef: ComponentStory<typeof ChipComponent> = (args) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const chipRef = React.useRef<BladeElementRef>(null);
+
+  return (
+    <Box gap="spacing.3" display="flex" flexDirection="column">
+      <ChipGroupComponent
+        accessibilityLabel="Select business type"
+        selectionType="single"
+        {...args}
+      >
+        <ChipComponent ref={chipRef} value="Proprietorship">
+          Proprietorship
+        </ChipComponent>
+        <ChipComponent value="Public">Public</ChipComponent>
+        <ChipComponent value="Small Business">Small Business</ChipComponent>
+      </ChipGroupComponent>
+      <Box maxWidth="200px">
+        <Button isFullWidth={false} onClick={() => chipRef?.current?.focus()}>
```

Comment:
shouldn't the chip be selected on focus when the selection type is single since its radio?

Also, can we add a button to reset the selection in this same story so that way we'll cover all the cases of refs and use cases too

Reply:
Yes, I'll update